### PR TITLE
Expand ci build & verify across all packages

### LINKED ISSUE #77: Standardise linting/CI checks across packages
Currently `feeds/npm` and `feeds/goproxy` have makefiles in place to run strict linting checks, linting/formatting strictness should probably be made consistent throughout the project.

### PR DESCRIPTION
**Pull request by tom--pollard**
Closes #77 

There may be a preference for having the lint commit before instead of after the commit in which the feature is enabled in CI, as such each commit passes. We may also want to tweak the `golangci.yml` file before committing the lint changes, as I don't know the provenance of it and it may require tailoring for the entire codebase.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ on:
   push:
   pull_request:
 jobs:
-  verify-terraform:
+  terraform-check:
```
Comment (Qinusty) [inline .github/workflows/ci.yml:7]: NIT: We should avoid bundling these changes with the lint commit

File: feeds/feed.go [hunk 2]
```diff
@@ -9,11 +10,13 @@ import (
 
 const schemaVer = "1.0"
 
+var errCutoff = errors.New("package was created before cutoff time")
```
Comment (Qinusty) [inline feeds/feed.go:13]: Can this be const?

File: feeds/nuget/nuget_test.go [hunk 3]
```diff
@@ -13,15 +13,19 @@ import (
...
 		"/v3/catalog0/page1.json": catalogPageMock,
 		"/v3/catalog0/data/somecatalog/new.expected.package.0.0.1.json": packageDetailMock,
 	}
-	srv := testutils.HttpServerMock(handlers)
-
-	testEndpoint, _ = url.Parse(srv.URL)
+	srv := testutils.HTTPServerMock(handlers)
+	testEndpoint, err = url.Parse(srv.URL)
+	if err != nil {
+		fmt.Println("Unexpected error during test url parsing: %w", err)
```
Comment (Qinusty) [inline feeds/nuget/nuget_test.go:27]: In a test this should probably be `t.Fatalf()`

File: feeds/nuget/nuget_test.go [hunk 4]
```diff
@@ -58,38 +62,61 @@ func indexMock(w http.ResponseWriter, r *http.Request) {
 	catalogEndpoint := makeTestURL("v3/catalog0/index.json")
 	response := fmt.Sprintf(`{"resources": [{"@id": "%s", "@type": "Catalog/3.0.0"}]}`, catalogEndpoint)
 
-	_, _ = w.Write([]byte(response))
+	_, err := w.Write([]byte(response))
+	if err != nil {
+		fmt.Println("Unexpected error during mock http server write: %w", err)
```
Comment (Qinusty) [inline feeds/nuget/nuget_test.go:67]: In a test this should probably be `t.Fatalf()` (duplicate)

File: feeds/nuget/nuget_test.go [hunk 5]
```diff
@@ -58,38 +62,61 @@ func indexMock(w http.ResponseWriter, r *http.Request) {
...
 	pageEndpoint := makeTestURL("v3/catalog0/page1.json")
 
-	response := fmt.Sprintf(`{"items": [{"@id": "%s", "commitTimeStamp": "%s"}]}`, pageEndpoint, time.Now().Format(time.RFC3339))
+	response := fmt.Sprintf(`{"items": [{"@id": "%s", "commitTimeStamp": "%s"}]}`,
+		pageEndpoint, time.Now().Format(time.RFC3339))
 
-	_, _ = w.Write([]byte(response))
+	_, err := w.Write([]byte(response))
+	if err != nil {
+		fmt.Println("Unexpected error during mock http server write: %w", err)
```
Comment (Qinusty) [inline feeds/nuget/nuget_test.go:79]: In a test this should probably be `t.Fatalf()` (duplicate)

File: feeds/nuget/nuget_test.go [hunk 6]
```diff
@@ -58,38 +62,61 @@ func indexMock(w http.ResponseWriter, r *http.Request) {
...
+	deletedItem := fmt.Sprintf(pkgTemplate,
+		makeTestURL("v3/catalog0/data/somecatalog/modified.not.expected.0.0.1.json"),
+		pkgDeleted, time.Now().UTC().Format(time.RFC3339))
 
 	response := fmt.Sprintf(`{"items": [%s, %s, %s]}`, addedItem, deletedItem, oldAddedItem)
 
-	_, _ = w.Write([]byte(response))
+	_, err := w.Write([]byte(response))
+	if err != nil {
+		fmt.Println("Unexpected error during mock http server write: %w", err)
```
Comment (Qinusty) [inline feeds/nuget/nuget_test.go:102]: In a test this should probably be `t.Fatalf()` (duplicate)

File: feeds/nuget/nuget_test.go [hunk 7]
```diff
@@ -58,38 +62,61 @@ func indexMock(w http.ResponseWriter, r *http.Request) {
...
 
 func packageDetailMock(w http.ResponseWriter, r *http.Request) {
-	response := fmt.Sprintf(`{"id": "new.expected.package", "version": "0.0.1", "published": "%s"}`, time.Now().UTC().Add(-1*time.Minute).Format(time.RFC3339))
+	response := fmt.Sprintf(`{"id": "new.expected.package", "version": "0.0.1", "published": "%s"}`,
+		time.Now().UTC().Add(-1*time.Minute).Format(time.RFC3339))
 
-	_, _ = w.Write([]byte(response))
+	_, err := w.Write([]byte(response))
+	if err != nil {
+		fmt.Println("Unexpected error during mock http server write: %w", err)
```
Comment (Qinusty) [inline feeds/nuget/nuget_test.go:112]: In a test this should probably be `t.Fatalf()` (duplicate)

File: feeds/nuget/nuget_test.go [hunk 8]
```diff
@@ -58,38 +62,61 @@ func indexMock(w http.ResponseWriter, r *http.Request) {
...
+	}
 }
 
 func makeTestURL(suffix string) string {
-	path, _ := url.Parse(suffix)
+	path, err := url.Parse(suffix)
+	if err != nil {
+		fmt.Println("Unexpected error during test url parsing: %w", err)
```
Comment (Qinusty) [inline feeds/nuget/nuget_test.go:119]: In a test this should probably be `t.Fatalf()` (duplicate)

File: feeds/packagist/packagist_test.go [hunk 9]
```diff
@@ -40,19 +43,60 @@ func TestFetch(t *testing.T) {
...
 	if r.URL.Query().Get("since") == "" {
 		w.WriteHeader(http.StatusBadRequest)
-		w.Write([]byte(`{"error":"Invalid or missing \u0022since\u0022 query parameter, make sure you store the timestamp at the initial point you started mirroring, then send that to begin receiving changes, e.g. https:\/\/packagist.org\/metadata\/changes.json?since=16145146222715 for example.","timestamp":16145146222715}`))
+		_, err := w.Write([]byte(`{"error":"Invalid or missing \u0022since\u0022 query parameter,
+		make sure you store the timestamp at the initial point you started mirroring,
+		then send that to begin receiving changes,
+		e.g. https:\/\/packagist.org\/metadata\/changes.json?since=16145146222715 for example."
+		,"timestamp":16145146222715}`))
+		if err != nil {
+			fmt.Println("Unexpected error during mock http server write: %w", err)
```
Comment (Qinusty) [inline feeds/packagist/packagist_test.go:52]: In a test this should raise the HTTP error similarly to the implementation of `HttpServe` outside of the tests

File: feeds/packagist/packagist_test.go [hunk 10]
```diff
@@ -40,19 +43,60 @@ func TestFetch(t *testing.T) {
...
+		"codeception/module-asserts":"^1.0.0","hoa/console":"^3.17"},
+		"support":{"issues":"https://github.com/ossf/package/issues",
+		"source":"https://github.com/ossf/package/tree/v1.0.0"}}]},"minified":"composer/2.0"}`,
 	}
 	for u, response := range m {
 		if u == r.URL.String() {
-			_, _ = w.Write([]byte(response))
+			_, err := w.Write([]byte(response))
+			if err != nil {
+				fmt.Println("Unexpected error during mock http server write: %w", err)
```
Comment (Qinusty) [inline feeds/packagist/packagist_test.go:98]: In a test this should raise the HTTP error similarly to the implementation of HttpServe outside of the tests (duplicate)

File: feeds/packagist/packagist_test.go [hunk 11]
```diff
@@ -40,19 +43,60 @@ func TestFetch(t *testing.T) {
...
+		,"timestamp":16145146222715}`))
+		if err != nil {
+			fmt.Println("Unexpected error during mock http server write: %w", err)
+		}
+	}
+	_, err := w.Write([]byte(`{"actions":[{"type":"delete","package":"to-delete/deleted-package",
+	"time":1614513806},{"type":"update","package":"ossf/package","time":1614514502},
+	{"type":"update","package":"ossf/package~dev","time":1614514502}],"timestamp":16145145025048}`))
+	if err != nil {
+		fmt.Println("Unexpected error during mock http server write: %w", err)
```
Comment (Qinusty) [inline feeds/packagist/packagist_test.go:59]: In a test this should raise the HTTP error similarly to the implementation of HttpServe outside of the tests (duplicate)

File: feeds/pypi/pypi_test.go [hunk 12]
```diff
@@ -71,4 +72,7 @@ func updatesXmlHandle(w http.ResponseWriter, r *http.Request) {
 	</channel>
 </rss>
 `))
+	if err != nil {
+		fmt.Println("Unexpected error during mock http server write: %w", err)
```
Comment (Qinusty) [inline feeds/pypi/pypi_test.go:76]: In a test this should raise the HTTP error similarly to the implementation of HttpServe outside of the tests (duplicate)

File: feeds/rubygems/rubygems_test.go [hunk 13]
```diff
@@ -119,4 +115,7 @@ func rubyGemsPackagesResponse(w http.ResponseWriter, r *http.Request) {
 ]
 
 `))
+	if err != nil {
+		fmt.Println("Unexpected error during mock http server write: %w", err)
```
Comment (Qinusty) [inline feeds/rubygems/rubygems_test.go:119]: In a test this should raise the HTTP error similarly to the implementation of HttpServe outside of the tests (duplicate)

File: feeds/feed.go (hunk 2 above)
Comment (tom--pollard) [reply feeds/feed.go:13]: I believe the convention is to use package level vars for wrapping errors with `Errorf` https://github.com/Djarvur/go-err113#reports

File: .github/workflows/ci.yml (hunk 1 above)
Comment (tom--pollard) [reply .github/workflows/ci.yml:7]: Agreed, missed this on the rebase

**Comment by tom--pollard** [comment]
@Qinusty I think I've addressed your comments in regards to raising `http.Error` / subsequent `t.Fatalf` in places where error handling was added, what do you think?

**Review by Qinusty — APPROVED** [review]
LGTM!

**Pull request merged**